app/controllers/facilities_controller.rb

```diff
@@ -12,4 +12,77 @@ def validate_params
   rescue ArgumentError
     raise Common::Exceptions::InvalidFieldValue.new('bbox', params[:bbox])
   end
+
+  private
+
+  def render_json(serializer, obj, options = {})
+    return render_collection(serializer, obj, options) if obj.class.name == 'ActiveRecord::Relation'
+
+    render_record(serializer, obj, options)
+  end
+
+  def render_collection(serializer, collection, options = {})
+    options = meta_pagination(collection, options)
+    render_record(serializer, collection, options)
+  end
+
+  def render_record(serializer, record, options = {})
+    render json: serializer.new(record, options)
+  end
+
+  def meta_pagination(paginated_obj, options = {})
+    options[:meta] = {} unless options.key?(:meta)
+    meta_options = options[:meta].merge(generate_pagination(paginated_obj))
+    options[:meta] = meta_options
+    options[:links] = {} unless options.key?(:links)
+    link_options = options[:links].merge(generate_links(paginated_obj))
+    options[:links] = link_options
+    options
+  end
+
+  def generate_pagination(paginated_obj)
+    {
+      pagination: {
+        current_page: paginated_obj.current_page,
+        prev_page: paginated_obj.previous_page,
+        next_page: paginated_obj.next_page,
+        total_pages: paginated_obj.total_pages
+      }
+    }
+  end
+
+  def generate_previus_page_link(paginated_obj)
```

Comment:
spelling issue, it looks consistent where it's called, but just want to point out

Reply:
I can't spell my way out of a wet paper bag.